src/compute/src/render/reduce.rs

```diff
@@ -2336,6 +2336,37 @@ mod monoids {
         }
     }
 
+    impl Clone for ReductionMonoid {
+        fn clone(&self) -> Self {
+            use ReductionMonoid::*;
+            match self {
+                Min(row) => Min(row.clone()),
+                Max(row) => Max(row.clone()),
+            }
+        }
+
+        // WARNING: Panics if self and source are of different `ReductionMonoid` enum entries.
+        fn clone_from(&mut self, source: &Self) {
+            use ReductionMonoid::*;
+            match (self, source) {
+                (Min(row), Min(source_row)) => {
+                    row.clone_from(source_row);
+                }
+                (Max(row), Max(source_row)) => {
+                    row.clone_from(source_row);
+                }
+                // FIXME(ptravers): not sure what we want to do here we can't overwrite
+                // without having owned access to the Row in dest. I don't see a way to
+                // achieve that without a clone. Is it better to panic or clone? I would lean
+                // clone for safety but I don't know how likely it is to hit the sad case
+                // in practice?
+                (dest, src) => soft_panic_or_log!(
+                    "Mismatched monoid variants in clone_from! destination: {dest:?} source: {src:?}",
+                ),
```

Comment:
this appears to be safe right now since the only usage of this function is in `plus_equals` which asserts that destination and source must be of the same enum variant

Reply:
Ah, I thought of something simpler, like this:
```
let mut row = match std::mem::take(self) { Min(row) | Max(row) => row };
let source_row = match source { Min(row) | Max(row) => row };
row.clone_from(source_row);
match other {
    Min(_) => *self = Min(row),
    Max(_) => *self = Max(row),
}
```

I'd prefer not to have surprising behavior built into the clone implementation, especially because it might surprise us in the future.

Reply:
ah I see! will refactor. that's much better!